### DISCUSSION TITLE
New `:rename` command

### DISCUSSION THREAD

**Discussion by nik-rev** [helix-editor/helix#12174] [Ideas]
You can use `:move` command to move the current buffer, but this is relative to Helix's current working directory.

Say your current working directory is `~/site/src/`, and you are editing the `~/site/src/lib/command.ts` file.

You want to rename it to `typed.ts`, you can do it with `:move lib/typed.ts`.

But it would be nice if we didn't need to specify the current file's directory. 

When https://github.com/helix-editor/helix/pull/11164 is merged, you'll be able to use `mv %{d}/typed.ts` but it would still be nice to have a `:rename` command., with a `:rn` alias.

**Comment by the-mikedavis** [comment]
I don't see us adding another command only to have such a small difference in behavior. Like you mentioned you should be able to use a variable in the future. Currently you can use `<C-r>%` to paste the current path and then `<C-w>` to remove the last segment, and then change the name.

**Comment by RoloEdits** [comment]
This seems like a good use case for flags: `:move --rename`

**Comment by baldwinn-anz** [comment]
related: https://github.com/helix-editor/helix/discussions/11616
Would be nice to be able to move based on a relative path to rename files.

e.g. for renaming `foo.ts` -> `bar.ts`
`:move ./bar.ts`

or for moving to a new folder (e.g. `foo.ts` -> `bar/foo.ts`)
`:move ./bar/foo.ts`
